tests/dataimportcron_test.go

```diff
@@ -103,11 +104,13 @@ var _ = Describe("DataImportCron", func() {
 			if i > 0 {
 				if createErrorDv {
 					By("Set desired digest to nonexisting one")
-					cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Get(context.TODO(), cronName, metav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					cron.Annotations[controller.AnnSourceDesiredDigest] = "sha256:12345678900987654321"
-					cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Update(context.TODO(), cron, metav1.UpdateOptions{})
-					Expect(err).ToNot(HaveOccurred())
+
+					//get and update!!!
+					Eventually(updateDataImportCron(f.CdiClient, ns, cronName,
+						func(cron *cdiv1.DataImportCron) *cdiv1.DataImportCron {
+							cron.Annotations[controller.AnnSourceDesiredDigest] = "sha256:12345678900987654321"
+							return cron
+						}), time.Second*30, time.Second).Should(BeNil())
```

Comment:
Not sure if I like it. Why should we fail on get/update if the controller works as expected and it and the test are the only ones updating of the CR? retrying it 30 times seems like sweeping bad controller behavior "under the rug".

Reply:
I think the idea is to not fail in case something else updated the DataImportCron in between us getting it, and trying to update. This is entirely possible, and since this is a rare failure, it could just be timing, and retrying it would make sense. Now retrying it 30 times seems a little overkill, if one retry is not enough it will likely be an issue in the controller if it is updating the object that much.

Reply:
right, 1 try should be enough ;)

Reply:
@arnongilboa @awels so what do you think, how to write this single retry in the easiest most readable way?
Use `Eventually` with default values, or with small timeout?

Reply:
introduced helper retryOnceOnErr, what do you think is this now good?

maybe we could improve it further.